lib/spack/spack/util/executable.py

```diff
@@ -233,6 +233,17 @@ def streamify(arg, mode):
         cmd_line_string = " ".join(escaped_cmd)
         tty.debug(cmd_line_string)
 
+        if self.exe[0].lower().startswith(("pwsh", "powershell", "cmd")):
+            spaced_args = [arg for arg in cmd if " " in arg]
+            cmd = " ".join([f'"{arg}"' if " " in arg else arg for arg in cmd])
+            if spaced_args:
+                tty.warn(
+                    "Spaces in command arguments can can confuse script parsing.",
```

Comment:
Awesome update, thanks! 

Can we add some comments describing why and when this is required and can we make this warning more specific to Windows/the shells as we have a similar warning above that could be easily confused with this one.